django_freeradius/base/models.py

```diff
@@ -710,7 +717,13 @@ def save(self):
                     check.value = getattr(profile, attr)
                     check.save()
                 else:
-                    radcheck.objects.create(username=self.user.username,
-                                            attribute=attribute_map.get(attr),
-                                            value=getattr(profile, attr))
+                    options = dict(username=self.user.username,
+                                   attribute=attribute_map.get(attr),
+                                   value=getattr(profile, attr))
+                    check = self._create_radcheck(**options)
+                    check.save()
         super(AbstractRadiusUserProfile, self).save()
+
+    def _create_radcheck(self, **kwargs):
```

Comment:
the method of this name is misleading because it does not create but it instantiates a new model instance.
To make it more readable you could do something like:

```python
@property
def _radcheck_model(self):
    return swapper.load_model('django_freeradius', 'RadiusCheck')
```

Reply:
@nemesisdesign the main reason for creating this method is to override this in openwisp-radius. i.e., add an extra organization field in the kwargs. Like https://github.com/openwisp/openwisp-radius/blob/master/openwisp_radius/models.py#L81

Reply:
ok just rename it then, the name is misleading